File: 01-OOP-Refresher/src/go/calculator.go
package main

// Go does NOT support method overloading.
// Use different function names instead.

func AddInt(a, b int) int          { return a + b }
func AddFloat(a, b float64) float64 { return a + b }
func AddThree(a, b, c int) int     { return a + b + c }

File: 01-OOP-Refresher/src/rust/calculator.rs
// Rust does NOT support function overloading.
// Use different method names or generics instead.

pub struct Calculator;

impl Calculator {
    pub fn add_i32(&self, a: i32, b: i32) -> i32 {
        a + b
    }

    pub fn add_f64(&self, a: f64, b: f64) -> f64 {
        a + b
    }

    pub fn add_three(&self, a: i32, b: i32, c: i32) -> i32 {
        a + b + c
    }
}
